datalist-polyfill.js

```diff
@@ -2,387 +2,389 @@
+          if (event.keyCode === keyESC) {
+            dataListSelect.setAttributeNode(document.createAttribute('hidden'));
+            dataListSelect.setAttribute('aria-hidden', 'true');
+
+            return;
+          }
+
+          var dataListOptions = dataList.querySelectorAll('option:not([disabled]):not(.message)'),
+            inputValue = eventTarget.value,
+            newSelectValues = document.createDocumentFragment(),
+            disabledValues = document.createDocumentFragment(),
+            visible = false,
+            multipleEmails = (eventTarget.type === 'email' && eventTarget.multiple),
+            keyOpen = (event.keyCode === keyUP || event.keyCode === keyDOWN);
+
+          // in case of type=email and multiple attribute, we would need to split the inputs value into pieces
+          if (multipleEmails) {
+            var multipleEntries = inputValue.split(','),
+              relevantIndex = multipleEntries.length - 1;
+
+            inputValue = multipleEntries[relevantIndex].trim();
+          }
+
+          // if the input contains a value, than ...
+          if (inputValue !== '' || keyOpen) {
+
+            // ... create an array out of the options list
+            var nodeArray = Array.prototype.slice.call(dataListOptions);
+
+            // ... sort all entries and
+            nodeArray.sort(function(a, b) {
+              return a.value.localeCompare(b.value);
+            })
+              .forEach(function(opt) {
+                var optionValue = opt.value;
+
+                // ... put this option into the fragment that is meant to get inserted into the select
+                // "Each option element that is a descendant of the datalist element, that is not disabled, and whose value is a string that isn't the empty string, represents a suggestion. Each suggestion has a value and a label." (W3C)
+                if (optionValue !== '' && optionValue.toLowerCase()
+                    .indexOf(inputValue.toLowerCase()) !== -1 && opt.disabled === false) {
+
+                  var label = opt.label,
+                    labelValueSeperator = ' / ',
+                    labelOptionPart = label.substr(0, optionValue.length + labelValueSeperator.length),
+                    optionPart = optionValue + labelValueSeperator;
+
+                  // the innertext should be value / label in case they are different
+                  if (label && label !== optionValue && labelOptionPart !== optionPart) {
+                    opt.innerText = optionValue + labelValueSeperator + label;
+
+                    // remove the label attribute, as it's being displayed differently on desktop and iOS Safari for our construct
+                    opt.setAttribute('data-label', opt.getAttribute('label'));
+                    opt.removeAttribute('label');
+                  } else // manipulating the option inner text, that would get displayed
```

Comment:
While you're formatting, maybe do the following, so the `else` and `if` aren't split up:
```
else if (!opt.innerText.trim()) {
    // manipulating the option ...
    // ...
}
```